### DISCUSSION TITLE
Nextcloud Security: Integrating Proxy Protocol (HAproxy L4)

### DISCUSSION THREAD

**Discussion by redyolocraft** [nextcloud/all-in-one#4438] [Ideas] [answered]
Hello,

First of all, I would like to thank everyone who worked on this project. This project is truly incredible. I have a layer 4 reverse proxy (HAProxy) and I wonder if it's possible to add in line 4 of the nextcloud.conf file of Apache (RemoteIPProxyProtocol On) because otherwise Nextcloud doesn't recognize the real IP address of the clients. Or maybe a small checkbox in the main container, because I understand that it's a potential vulnerability if, for example, a client manages to send a custom TCP packet directly to Apache, but for me, it's on an isolated network.

**Comment by szaimen** [comment]
> This project is truly incredible. 

Thank you! ☺️

---

Hi, I don't know if it is that easy to simply do RemoteIPProxyProtocol On since caddy always sits between apache and external reverse proxies. Maybe this would need to be enabled in caddy as well?

Some help on this is appreciated :)

**Reply by szaimen** [reply]
It looks like this should be possible with caddy: https://caddyserver.com/docs/caddyfile/options#proxy-protocol PRs are welcome!

**Comment by szaimen** [comment, marked as answer]
There is a WIP now here: https://github.com/szaimen/aio-caddy/pull/100 and https://github.com/nextcloud/all-in-one/pull/7191

**Reply by szaimen** [reply]
This is now released with v12.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
